scripts/generate_mrenclave.sh

```diff
@@ -0,0 +1,43 @@
+#!/bin/bash
+
+
+# Copyright 2020 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+if [ "$1" == "" ]; then
+    echo 'Parse the output of the SGX tool to extract the MRENCLAVE value'
+    echo 'and save it to file $TCF_HOME/metadata_info.txt by default'
+    echo 'Usage: generate_mrenclave.sh <library location> <output file(optional)>'
+    exit
+fi
+
+LIB="$1"
+if [ -z "$2" ];then
+	OUT_FILE="${TCF_HOME}/metadata_info.txt"
+else
+	OUT_FILE=$2
+fi	
+
+rm -f ${OUT_FILE}
+
+sgx_sign dump -enclave ${LIB} -dumpfile metadata.txt
+
+awk '/metadata->enclave_css.body.enclave_hash.m:/{flag=1; next} 
+    /metadata->enclave_css.body.isv_prod_id:/{flag=0} flag' metadata.txt  > temp_file.txt
+    
+awk '!a[$0]++' temp_file.txt > ${OUT_FILE}
+
+rm -f metadata.txt
+rm -f temp_file.txt
```

Comment:
Can we avoid generating temporary files. You could use pipes to connect the commands or store the output of the commands in variables

Reply:
Generation of metadata.txt is must because  sgx_sign command requires a file to dump metadata information (text format). 
Pipe can be used to connect 2 awk command and avoid temp_file.txt generation.

Reply:
Done